Src/ILGPU/Backends/PTX/PTXKernelFunctionGenerator.cs

```diff
@@ -260,19 +261,19 @@ private void SetupKernelIndex(Register indexRegister, Register lengthRegister)
             {
                 EmitImplicitKernelIndex(
                     0,
-                    indexRegister as PrimitiveRegister,
-                    lengthRegister as PrimitiveRegister);
+                    (indexRegister as PrimitiveRegister).AsNotNull(),
+                    (lengthRegister as PrimitiveRegister).AsNotNull());
             }
             else
             {
-                var compoundIndex = indexRegister as CompoundRegister;
-                var compoundLength = lengthRegister as CompoundRegister;
+                var compoundIndex = (indexRegister as CompoundRegister).AsNotNull();
```

Comment:
It is already implemented as `AsNotNull<T>`. Maybe we can add a new `CastAsNotNull<T>` extension?

Reply:
This would be nice! I think this can simplify the code significantly, as `(xyz as T).AsNotNull` becomes `xyz.CastAsNotNull<T>()` or `xyz.NotNullCast<T>()`.